pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -459,6 +459,7 @@ func (configgen *ConfigGeneratorImpl) buildGatewayHTTPRouteConfig(node *model.Pr
 				gatewayRoutes[gatewayName][vskey] = routes
 			}
 			for _, hostname := range intersectingHosts {
+				hostname = host.Name(strings.ToLower(hostname.String()))
```

Comment:
general question -- is this the only place we are handling hostnames where case matters?

Seems like we have possible 100s of places. Maybe we need some very high level tests exercising a variety of areas

Reply:
At least for routing these are the only place. But there may be others. Need to check. Agree on high level tests for a variety of areas